sway/input/keyboard.c

```diff
@@ -603,8 +608,23 @@ void sway_keyboard_configure(struct sway_keyboard *keyboard) {
 		}
 	}
 
+	bool keymap_changed = false;
+	bool effective_layout_changed = keyboard->effective_layout != 0;
```

Comment:
Hmm, not sure I understand the logic here...

Reply:
Hmm, nevermind. It's the effective layout that gets reset.

However if the keymap changed the effective layout changes too, even if `effective_layout` is still zero. The effective layout depends on the keymap.

Reply:
Yes, but if the effective layout is 0 before and after, then either the entire keymap changed or the keymap is identical and there is no change to the effective layout. See #4343 for a follow-up for this